### ISSUE TITLE
ABCOP and MCAI as a variant of COP and SCAI

### ISSUE THREAD

**Issue by gmandorl** [cloudsci/cloudmetrics#78]
The index ABCOP is a modified version of COP (not so similar though), and MCAI is a modified version of SCAI.
There are two ways to implement them in the code:
  - as a variation of COP and SCAI, similarly to issue #57 for Iorg.
  - each one independently, like a new index, i.e. creating the new files `mcai.py` and `abcop.py` .
 
What do you think is the best?

$\textrm{COP} = \frac{1}{N(N-1)} \sum_{i} \sum_{j\neq i} \frac{\sqrt{A_i/\pi} + \sqrt{A_j/\pi}}{d(i,j)}$
$\textrm{ABCOP} = \frac{1}{2L} \sum_{i} \max_{j\neq i} \left( \frac{A_i + A_j}{d_2(i,j)} \right)$
Differences between COP and ABCOP:
 - sum over $i$ instead of mean over $i$
 - max over $j$ instead of mean over $j$
 - Area $A_i$ instead of equivalent radius $\sqrt{A_i/\pi}$
 - distances $d_2(i,j) = d(i,j) - \frac{1}{2}(\sqrt{A_i/\pi} + \sqrt{A_j/\pi})$ instead of centroid distances $d(i,j)$

$\textrm{SCAI} = \frac{N}{N_{max}} \frac{D_0}{L}$    where $D_0$ is the geometric mean of $d(i,j)$
$\textrm{MCAI} = \frac{N}{N_{max}} \frac{D_2}{L}$    where $D_2$ is the arithmetic mean of $d_2(i,j)$
The only difference is $D_0$ -> $D_2$

ABCOP = Area-Based COP [published here](https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2022JD036665)
MCAI    = Modified Convective Aggregation Index [published here](https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2019JD030816)

**Comment by observingClouds** [comment]
I would give each of your indices a new function name and wouldn't integrate them into the existing functions as options. The reasoning behind this is that you have unique names for each algorithm/index. Integrating your indices as options to the existing ones might lead to confusion/less specificity. In case of `Iorg` it is often not further defined in manuscripts which implementation is being used leading to confusion.

**Comment by martinjanssens** [comment]
Hm, I'm agnostic to this. On one hand, I agree @observingClouds , and on the other, abcop and mcai aren't really new metrics, they're just new variants of existing metrics, which is more directly reflected in a single function for a 'family' of metrics. I'd be okay with doing separate functions in separate files, with the docstring of `abcop.py` mentioning explicitly that it's just area-weighted `cop` , and similar for `mcai` (as you've already done above, @gmandorl).

**Comment by gmandorl** [comment]
Great! I'll create individual files for each, and I'll make it clear in the docstrings that they are variations of COP and SCAI.

**Comment by leifdenby** [comment]
> with the docstring of `abcop.py` mentioning explicitly that it's just area-weighted `cop` , and similar for `mcai` (as you've already done above, @gmandorl).

Yes, I would opt for this too. Separate functions (since they have separate names in published literature).

Great work here @gmandorl !

**Comment by gmandorl** [comment]
I have a doubt about SCAI.


At
https://github.com/cloudsci/cloudmetrics/blob/master/cloudmetrics/objects/metrics/scai.py#L84
scai is defined by
$\textrm{SCAI} = \frac{N}{N_{max}} \frac{D_0}{L}$ where $N_{max} = nx * ny / 2$ and L=1000 is an external parameter


However, from [Tobin et al. (2012)](https://journals.ametsoc.org/view/journals/clim/25/20/jcli-d-11-00258.1.xml): "the distances  $D_0$ or $D_1$ are normalized by the characteristic length L (here, about 1000 km) of the domain under consideration".
Therefore, I was thinking that L=sqrt(nx * ny) and it is not an external parameter.


What do you think? Should we change L to sqrt(nx * ny), or we should keep it fixed?

**Comment by martinjanssens** [comment]
In our debates about SCAI (see #42), we essentially settled on leaving it up to the user. We leave L as a reference length that can be provided as an optional input (`reference_lengthscale`), whose default is 1000, but which can be set to something else, e.g. `sqrt(nx * ny)`. I think I still stand by that choice, but don't have strong feelings either way. Most important to me is that the documentation clearly says that choosing the length scale is somewhat arbitrary - do you think it does?
